envisage/plugin.py

```diff
@@ -289,7 +289,13 @@ def unregister_services(self):
         service_ids.reverse()
 
         for service_id in service_ids:
-            self.application.unregister_service(service_id)
+            # note the service may have already been individually unregistered
+            try:
+                self.application.service_registry.get_service_from_id(service_id)  # noqa: E501
```

Comment:
I suspect Plugin is not supposed to use `service_registry` on the `application` directly, and it would be more symmetric if we call `self.application.get_service_from_id` here instead before we call `self.application.unregister_service`. (I know at the end of the day they are the same after all the redirection. It just looks asymmetric and inconsistent...)